### PR TITLE
Ticket 1: Add app configuration

### PR DESCRIPTION
**Pull request by eyal0226**

### CODE REVIEW

File: src/components/Product_card/ProductCard.tsx [hunk 1]
```diff
@@ -0,0 +1,29 @@
+import React from "react";
+
+interface ProductCard_props {
```
Comment (bar-cohen-1) [inline src/components/Product_card/ProductCard.tsx]: Rename to `ProductCardProps`

File: package.json [hunk 2]
```diff
@@ -3,24 +3,30 @@
...
     "@types/node": "^16.18.11",
     "@types/react": "^18.0.26",
     "@types/react-dom": "^18.0.10",
+    "axios": "^1.2.2",
     "react": "^18.2.0",
     "react-dom": "^18.2.0",
+    "react-redux": "^8.0.5",
     "react-scripts": "5.0.1",
-    "typescript": "^4.9.4",
+    "typescript-eslint": "^0.0.1-alpha.0",
```
Comment (bar-cohen-1) [inline package.json]: This library is unstable. Remove it or change to stable version

File: package.json [hunk 3]
```diff
@@ -39,5 +45,31 @@
...
+    "eslint-plugin-prettier": "^4.2.1",
+    "eslint-plugin-react": "^7.31.11",
+    "eslint-plugin-react-hooks": "^4.6.0",
+    "husky": "^8.0.0",
+    "lint-staged": "^13.1.0",
+    "prettier": "^2.8.1",
+    "typescript": "^4.9.4"
+  },
+  "lint-staged": {
+    "*.tsx": [
```
Comment (bar-cohen-1) [inline package.json]: also for `.ts`

File: src/App.tsx [hunk 4]
```diff
@@ -1,24 +1,17 @@
-import React from 'react';
-import logo from './logo.svg';
-import './App.css';
+import React from "react";
+import "./App.css";
+import Product_card from "./components/Product_card/ProductCard";
```
Comment (bar-cohen-1) [inline src/App.tsx]: Use absolute paths and rename to ProductCard

File: package.json [hunk 5]
```diff
@@ -39,5 +45,31 @@
...
+      "npm run lint",
+      "npm run format"
+    ],
+    "*.test.tsx": [
```
Comment (bar-cohen-1) [inline package.json]: same here

File: src/components/Product_card/ProductCard.tsx [hunk 6]
```diff
@@ -0,0 +1,29 @@
+import React from "react";
```
Comment (amitmarko) [inline src/components/Product_card/ProductCard.tsx]: Change folder name to `ProductCard` and the file to `index.tsx`

File: src/components/Product_card/ProductCard.tsx [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+      <p>{name}</p>
+      <p>{brand}</p>
+      <p>{price}$</p>
+      <p>{id}</p>
+      {product_image && <p>{product_image.url}</p>}
+    </div>
+  );
+};
+
+export default ProductCard;
```
Comment (amitmarko) [inline src/components/Product_card/ProductCard.tsx]: Use regular export instead 
```suggestion
```

File: src/components/Product_card/ProductCard.tsx [hunk 8]
```diff
@@ -0,0 +1,29 @@
...
+
+interface ProductCard_props {
+  name: string;
+  brand: string;
+  product_image?: { url: string };
+  price: number;
+  id: number;
+}
+
+const ProductCard = ({
```
Comment (amitmarko) [inline src/components/Product_card/ProductCard.tsx]:
```suggestion
export const ProductCard = ({
```

File: src/components/Product_card/ProductCard.tsx [hunk 9]
```diff
@@ -0,0 +1,29 @@
...
+  name,
+  brand,
+  price,
+  id,
+  product_image,
+}: ProductCard_props) => {
+  return (
```
Comment (amitmarko) [inline src/components/Product_card/ProductCard.tsx]: Remove return

**Comment by amitmarko** [comment]
@eyal0226 change PR name to: `Ticket 1: Add app configuration`

File: src/components/Product_card/ProductCard.tsx [hunk 10]
```diff
@@ -0,0 +1,25 @@
+import React from "react";
+
+interface ProductCard_props {
+  name: string;
+  brand: string;
+  product_image?: { url: string };
```
Comment (amitmarko) [inline src/components/Product_card/ProductCard.tsx]: You already on product object therefore no need add `product` pre-fix
```suggestion
  image?: { url: string };
```

File: src/components/Product_card/ProductCard.tsx [hunk 11]
```diff
@@ -0,0 +1,25 @@
...
+  price: number;
+  id: number;
```
Comment (amitmarko) [inline src/components/Product_card/ProductCard.tsx]: Should be string
```suggestion
  id: string;
```

File: src/components/Product_card/ProductCard.tsx [hunk 12]
```diff
@@ -0,0 +1,25 @@
...
+}
+
+export const ProductCard = ({
+  name,
+  brand,
+  price,
+  id,
+  product_image,
+}: ProductCard_props) => {
+  <div className="card_cont">
```
Comment (amitmarko) [inline src/components/Product_card/ProductCard.tsx]: Don't use classNames or css, Styled Component 
```suggestion
  <Root>
```

File: src/components/Product_card/ProductCard.tsx [hunk 13]
```diff
@@ -0,0 +1,25 @@
...
+    <p>{name}</p>
+    <p>{brand}</p>
+    <p>{price}$</p>
+    <p>{id}</p>
```
Comment (amitmarko) [inline src/components/Product_card/ProductCard.tsx]: No need showing id 
```suggestion
```

Comment (eyal0226) [reply src/components/Product_card/ProductCard.tsx]: it was just because of errors that TS gave me

**Review by bar-cohen-1 — APPROVED** [review]

**Review by bar-cohen-1 — APPROVED** [review]

**Pull request merged**